.github/workflows/macos-generation.yml

```diff
@@ -0,0 +1,198 @@
+name: macOS image generation
```

Comment:
Should we move reusable steps to the separate folder? `.github/workflows/steps`?

Reply:
It looks like reusable workflows must still reside in the .github/workflows directory
https://github.com/actions/runner/issues/1493

Reply:
Oh, you are right. Only reusable actions can be moved to different folder 😢